mocha third party reporter

I followed the [wiki ](https://github.com/visionmedia/mocha/wiki/Third-party-reporters) to create a reporter. Then i published it to npm and included it in my project as devDependency, but i get an error after running mocha with the reporter:

``` javascript
$ mocha --reporter xunit-file

module.js:340
    throw err;
          ^
Error: Cannot find module './reporters/xunit-file'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:362:17)
    at require (module.js:378:17)
    at Mocha.reporter (MOCHA_ROOT/lib/mocha.js:93:20)
    at Object.<anonymous> (MOCHA_ROOT/bin/_mocha:176:7)
    at Module._compile (module.js:449:26)
    at Object.Module._extensions..js (module.js:467:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.runMain (module.js:492:10)
    at process.startup.processNextTick.process._tickCallback (node.js:244:9)
```

Seems like mocha is only looking in the `./reporters/` folder. When i am copying the reporter in this folder everything works.

--- a/lib/mocha.js
+++ b/lib/mocha.js
@@ -94,7 +94,11 @@ Mocha.prototype.reporter = function(reporter){
     this._reporter = reporter;
   } else {
     reporter = reporter || 'dot';
-    this._reporter = require('./reporters/' + reporter);
+    try {
+      this._reporter = require('./reporters/' + reporter);
+    } catch (err) {
+      this._reporter = require(reporter);
+    }
     if (!this._reporter) throw new Error('invalid reporter "' + reporter + '"');
   }
   return this;

